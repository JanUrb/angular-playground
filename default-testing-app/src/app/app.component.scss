app-submit{
  max-width: 500px;
  margin: auto;
  justify-content: center;
}