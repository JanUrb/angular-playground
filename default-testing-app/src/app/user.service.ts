import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root',
})
export class UserService {
  constructor() {}

  userCache: User[] = [User.build("Jan", 182)];

  public async loadCachedUser(): Promise<User[]>{
    return this.userCache;
  }

  public loadCachedUserBlocking(): User[]{
    return this.userCache;
  }
}

export class User {
  name: string;
  height: number;

  static build(name, height): User {
    return { name, height};
  }
}
