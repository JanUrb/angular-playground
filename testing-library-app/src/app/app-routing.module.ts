import { SummaryResolver } from '@angular/compiler';
import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';
import { User } from './user.service';

const routes: Routes = [];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule],
})
export class AppRoutingModule {}
