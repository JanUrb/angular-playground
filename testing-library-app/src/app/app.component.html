<h1>{{title}} is running!</h1>

<app-submit (addUser)="addUser($event)"></app-submit>

<div>
  <h2>User:</h2>
  <ul>
    <li *ngFor="let user of users">
        <p>Name: {{user.name}}</p>
        <p>Größe: {{user.height}}</p>
    </li>
  </ul>
</div>


<router-outlet></router-outlet>