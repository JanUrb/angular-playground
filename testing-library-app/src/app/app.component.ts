import { Component, Input, OnInit } from '@angular/core';
import { User, UserService } from './user.service';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss'],
})
export class AppComponent implements OnInit {
  
  title = 'default-testing-app';

  users: User[] = [];

  constructor(private userService: UserService) {}
  ngOnInit(): void {
    this.userService.loadCachedUser().then((user) => {
      this.users.push(...user);
    });
  }

  addUser(user: User): void {
    this.users.push(user);
  }
}
