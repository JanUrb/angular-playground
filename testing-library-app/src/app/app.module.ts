import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { ReactiveFormsModule } from '@angular/forms';
import { CommonModule } from '@angular/common';
import { SubmitComponent } from './submit/submit.component';

@NgModule({
  declarations: [
    AppComponent,
    SubmitComponent
  ],
  imports: [
    BrowserModule,
    CommonModule,
    AppRoutingModule,
    ReactiveFormsModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
