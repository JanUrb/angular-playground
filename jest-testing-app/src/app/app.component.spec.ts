import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import {
  TestBed,
  async,
  tick,
  fakeAsync,
  waitForAsync,
} from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { AppComponent } from './app.component';
import { User, UserService } from './user.service';

describe('AppComponent', () => {
  let userServiceSpy;

  beforeEach(waitForAsync(() => {
    userServiceSpy = {loadCachedUser: jest.fn() }
    userServiceSpy.loadCachedUser.mockReturnValueOnce(
      Promise.resolve([User.build('Jan', 182)])
    );

    TestBed.configureTestingModule({
      imports: [RouterTestingModule],
      declarations: [AppComponent],
      providers: [{ provide: UserService, useValue: userServiceSpy }],
      schemas: [CUSTOM_ELEMENTS_SCHEMA]
    }).compileComponents();
  }));

  it('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.componentInstance;
    expect(app).toBeTruthy();
  });

  it(`should have as title 'default-testing-app'`, () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.componentInstance;
    expect(app.title).toEqual('default-testing-app');
  });

  it('should render title', () => {
    const fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges();
    const compiled = fixture.nativeElement;
    expect(compiled.querySelector('h1').textContent).toContain(
      'default-testing-app is running!'
    );
  });

  it('should show cached user', fakeAsync(() => {
    const fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges();
    const compiled = fixture.nativeElement;
    tick(1);
    fixture.detectChanges(); //=> was wird hier getestet? Die Komponente oder Angular?
    expect(compiled.querySelector('li').textContent).toContain(
      'Jan'
    );
  }));

  
it('should render correctly', () => {
  const fixture = TestBed.createComponent(AppComponent);
  expect(fixture).toMatchSnapshot(); // Schritt 1: Legt den Snapshot an, Schritt 2 vergleicht snapshots
});

});
