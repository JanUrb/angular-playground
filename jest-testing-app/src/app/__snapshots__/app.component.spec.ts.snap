// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should render correctly 1`] = `
<app-root
  title={[Function String]}
  userService={[Function Object]}
  users={[Function Array]}
>
  <h1>
    
  </h1><app-submit /><div>
    <h2>
      User:
    </h2>
    <ul>
      
    </ul>
  </div><router-outlet />
</app-root>
`;
